### ISSUE TITLE
"Request Entity Too Large" - Can't save draft

### ISSUE THREAD

**Issue by metallikat36** [dsifford/academic-bloggers-toolkit#271]
ABT Version: 4.7.1
PHP Version: 5.6.29
Theme: Dazzling
Browser: Old unsupported Chrome on Vista

Plugin List:
Academic Blogger's Toolkit
Ajax Search Lite
All In One WP Security
CM Table Of Contents Pro
Disqus Conditional Load
Easy Google Fonts
Give - Donation Plugin
Hyper Cache
Jetpack
Popular Posts Tabbed Widget for Jetpack
PopupAlly
Scalable Vector Graphics (SVG)
Social Popscroll
Super Socializer
UpdraftPlus - Backup/Restore
WP Smush
Yoast SEO

I am inputting a post that is lroughly 50 pages long into wordpress. It was gonna have like 400 references, but as I got to inputting around number 140, when I try to save the draft I get the following error: ""Request Entity Too Large"

![413 request entity too large](https://cloud.githubusercontent.com/assets/17968340/21379285/b040f7f2-c701-11e6-85aa-e12d539effce.png)

I contacted Siteground (my host) and they increased the variables "LimitRequestBody" and "post_max_size". That didn't fix it. So I think maybe I did something to break the ABT plugin. Any ideas? It's possible I might have unknowingly reordered some references. The only strange behaviors I noted were as follows:

1. A few times I saw that after adding an inline citation, the post added a paragraph break (i.e. a blank line) before the start of the paragraph that the inline citation belonged to. I would then just delete the extra line and think nothing of it. 
2. When I tried rolling back to earlier version, the references list doesn't roll back. Like if I have 148 references and then rollback to a time when there was only 140 references, the references will still have 148 references. And if those extra references (i.e. 141 to 148) were used as inline citations, but now no longer have the text present in the rolled-back version, then those citations seemed to default to going to the very beginning of the post. And it would retain the indices "141-148", even though my style should have renumbered them. Though I actually didn't try refreshing. 

Anyway, long story short, I am just going to start a new post and try and input it all from scratch being careful not to reorder anything. It would be great if there was a way I could copy/paste form my current broken post to a new post so I don't have to reinput all the references, but that's life.

Thanks,
David

**Comment by dsifford** [comment]
Hi there,

If I might ask, what are you writing that is 50 pages long on wordpress? Seems like a huge amount of content for a single blog post!

Unfortunately, your issue is not related at all to this plugin. It's entirely because of how wordpress manages saving your post content to the database. It's done in a single HTTP POST request which, as you might have guessed, runs out of memory quickly.

You might want to try ramping up your [`max_execution_time`](http://php.net/manual/en/info.configuration.php#ini.max-execution-time) to see if that works for you. You'll also have to _really_ ramp up your `post_max_size` as siteground suggested. It needs to be huge to handle that much data.

Re: references not rolling back -- That's a necessary evil. If I were to add in additional caching each time a post is saved, that would exponentially increase the size of the saved data and wouldn't be useful in 99% of the cases.

Closing this issue because it's out of scope (server issue).

**Comment by metallikat36** [comment]
What am I writing? An article refuting the garbage science behind antidepressants. If you are a med student, you should definitely read what I have so far:
http://www.earthmedresearch.org/antidepressants-fraudulent-depression-complex-environmental-disease/
The greater the ignorance about a topic, the more pages it takes to unwind that ignorance.

Thanks, I'll try those other fixes.

**Comment by metallikat36** [comment]
These are the maximum limits Siteground allows:

max_execution_time = 120
max_input_time = 120
memory_limit = 768M
post_max_size = 128M
upload_max_filesize = 128M

This all looks like they should be big enough to handle that post, don't they?

**Comment by dsifford** [comment]
Not sure. PHP and I don't really get along, so unfortunately I won't be able to be very useful for this issue.

Perhaps you could just chunk the post into a few smaller posts? The benefit there also would be that it would also be more digestible by the reader.
